psycopg_pool/psycopg_pool/base.py

```diff
@@ -94,9 +94,7 @@ def __init__(
         # connections to the pool.
         self._growing = False
 
-        # _close should be the last property to be set in the state
-        # to avoid warning on __del__ in case __init__ fails.
-        self._closed = False
+        self._closed = True
```

Comment:
I'd say so, as expected from context manager.

Reply:
Well, not necessarily. Once a connection or cursor are closed they can't be reopened.

If this behaviour is expected we should add tests for it and verify that the main functionalities (the background workers) are up again as expected. If it's not a feature we want to support (and it might make our life easier) then we should explicitly forbid it.

Reply:
Gone with the restart option as it's easily achievable and it makes state handling a little more consistent in my opinion.